src/nxt_http_parse.c

```diff
@@ -1182,6 +1182,7 @@ nxt_http_fields_hash(nxt_lvlhsh_t *hash,
 
     lhq.replace = 0;
     lhq.proto = &nxt_http_fields_hash_proto;
+    lhq.pool = NULL;
 
     for (i = 0; i < count; i++) {
         key = NXT_HTTP_FIELD_HASH_INIT;
```

Comment:
I don't think this is needed...

lhq.proto.alloc  = nxt_lvlhsh_alloc

Which is

```c
void *
nxt_lvlhsh_alloc(void *data, size_t size)
{
    return nxt_memalign(size, size);
}
```

Where lhq.pool (data) is *not* used.

Even with this

```c
nxt_lvlhsh_convert_bucket_to_level()
{
    ...
    proto = lhq->proto;
    ...
    q.proto = proto;                                                        
    q.pool = lhq->pool;
    ...
}
```

lhq.pool is *not* used with nxt_lvlhsh_alloc()

It's also not used by

```c
void                                                                            
nxt_lvlhsh_free(void *data, void *p)                                                                            
{                                                                               
    nxt_free(p);                                                                
}
```

Reply:
I get it. `lhq->pool` is passed through these functions for unbeknownst reason.

Ah, so is it a bug in [MSan](https://clang.llvm.org/docs/MemorySanitizer.html)?
Not actually, MSan gives the responsibility to the `caller` for initializing all the parameters for the `callee`.

This particular MSan error is a [fuzz-blocker](https://firefox-source-docs.mozilla.org/tools/fuzzing/index.html#fuzz-blockers) for `fuzz_http`.

#### There are currently three fixes.

* The best, of course is to merge this PR: https://github.com/nginx/unit/pull/1295.
* Disabling `fuzz_http` for `MSan`, I don't think that is a good idea.
* Run this PR as a patch in `oss-fuzz.sh` compiling script, or run `disable_msn.patch` patch in `oss-fuzz.sh`.

disable_msn.patch
```patch
index 7a8b3dd..025827b 100644
--- a/src/nxt_lvlhsh.c
+++ b/src/nxt_lvlhsh.c
@@ -284,7 +284,7 @@ nxt_lvlhsh_insert(nxt_lvlhsh_t *lh, nxt_lvlhsh_query_t *lhq)
 }
 
 
-static nxt_int_t
+__attribute__((no_sanitize("memory"))) static nxt_int_t
 nxt_lvlhsh_new_bucket(nxt_lvlhsh_query_t *lhq, void **slot)
 {
     uint32_t  *bucket;
@@ -448,7 +448,7 @@ nxt_lvlhsh_bucket_insert(nxt_lvlhsh_query_t *lhq, void **slot, uint32_t key,
 }
 
 
-static nxt_int_t
+__attribute__((no_sanitize("memory"))) static nxt_int_t
 nxt_lvlhsh_convert_bucket_to_level(nxt_lvlhsh_query_t *lhq, void **slot,
     nxt_uint_t nlvl, uint32_t *bucket)
 {
````

Also a very simple question: Does this initialization of value break anything in this code-base?

Reply:
Ah, this PR is like `c standards` or `-Wextra`. It's not important, but it's good to use it.